In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("../data/student_productivity_cleaned.csv")
df.head()

,age,academic_level,sleep_hours,screen_time_hours,exercise_minutes,caffeine_intake_mg,part_time_job,upcoming_deadline,internet_quality,mental_health_score,focus_index,burnout_level,productivity_score,exam_score,study_hours_total,leisure_screen_time
0,18,0.0,6.52,6.47,81,38,1,0,2.0,10,43.05,31.77,73.65,50.16,11.40,5.24
1,18,0.0,5.97,6.05,111,339,0,0,2.0,3,15.92,37.00,13.70,1.00,6.53,4.20
2,22,0.0,8.39,7.62,68,266,0,1,2.0,8,27.39,34.37,45.15,18.30,3.74,3.42
3,17,0.0,6.31,11.67,113,480,1,1,0.0,3,22.31,77.31,20.92,9.37,10.84,4.47
4,19,0.0,8.01,10.02,121,24,1,0,2.0,8,38.11,39.53,59.23,27.81,11.88,3.35


In [4]:
#Como é o aluno médio deste dataset
df[['age','sleep_hours', 'study_hours_total','leisure_screen_time','exam_score']].median()

age                    20.000
sleep_hours             7.010
study_hours_total       9.030
leisure_screen_time     4.535
exam_score             18.010
dtype: float64

# O que realmente impacta a nota final?
- É comum pensar que uma maior quantidade de horas estudadas garante o sucesso acadêmico. Para testar essa hipótese, utilizei a **correlação de Pearson** para descobrir quais variáveis têm uma maior relação com a nota final

In [5]:
# checando a relação de cada variável com as notas

correlacao = pd.DataFrame(df.corr()['exam_score'])
correlacao = correlacao.sort_values(by='exam_score',ascending=False)

In [6]:
correlacao_positiva = correlacao.drop('exam_score')
print(f'Fatores com maior impacto positivo nas notas:\n {correlacao_positiva.head(5)}')

Fatores com maior impacto positivo nas notas:
                      exam_score
productivity_score     0.886401
focus_index            0.749879
mental_health_score    0.546547
study_hours_total      0.436046
sleep_hours            0.234710


# Sobres os impactos positivos
- Fatores como produtividade, foco e saúde mental possuem um impacto considerável no sucesso acadêmico de um estudante, superando até mesmo as horas brutas estudadas.

- **Quantidade vs Qualidade**: A análise de correlação mostra que estudar com a mente cansada e sem foco rende menos do que estudar por poucas horas, mas com qualidade

In [7]:
print(f'Fatores com maior impacto negativo nas notas:\n {correlacao.tail(5).sort_values(by="exam_score",ascending=True)}')

Fatores com maior impacto negativo nas notas:
                      exam_score
burnout_level         -0.407808
upcoming_deadline     -0.215342
part_time_job         -0.149807
screen_time_hours     -0.131940
leisure_screen_time   -0.116934


# Sobre os impactos negativos
- Notavelmente, o fator que gera o maior impacto negativo é nível de burnout dos estudantes

- Curiosamente, o tempo que o estudante gassta em telas (redes socias e jogos) tem pouco impacto na sua performance final

- Conclusão: A sobrecarga mental afeta muito mais o desempenho de um estudante do que fatores externos como o tempo de lazer

## 3. Investigando o Vilão: O que causa o Burnout?
Na análise anterior, descobrimos que o nível de esgotamento (burnout) é o principal inimigo do desempenho acadêmico. Mas o que leva um estudante ao burnout? 

Para entender isso, vamos isolar a variável `burnout_level` e procurar suas maiores correlações. O que tem correlação **positiva** aumenta o esgotamento. O que tem correlação **negativa** age como um "aliado" contra o esgotamento mental.

In [8]:
correlacao_burnout = pd.DataFrame(df.corr()['burnout_level']).drop('burnout_level')
correlacao_burnout = correlacao_burnout.sort_values(by='burnout_level', ascending=False)

print("Fatores que mais AUMENTAM o Burnout:")
print(correlacao_burnout.head(5))

Fatores que mais AUMENTAM o Burnout:
                    burnout_level
upcoming_deadline        0.510612
part_time_job            0.372089
screen_time_hours        0.340652
caffeine_intake_mg       0.197278
internet_quality         0.010324


In [9]:
print("Fatores que mais REDUZEM o Burnout:")
print(correlacao_burnout.tail(5).sort_values(by='burnout_level', ascending=True))

Fatores que mais REDUZEM o Burnout:
                    burnout_level
sleep_hours             -0.492935
exam_score              -0.407808
productivity_score      -0.267304
exercise_minutes        -0.157311
focus_index             -0.075425


> ## Insights da Análise de Burnout:
> 
> * **Impactos negativos:** A pressão de **Prazos Próximos** (0.51) é, de longe, o maior causador de estresse, seguido por tentar conciliar os estudos com um **Trabalho de Meio Período**(0.37). Outro fator interessante é que o alto tempo de **exposição geral a telas**(0.34) e o alto consumo de **Cafeína** (0.19) também contribuem para o esgotamento.
> 
> * **O que ajudam a reduzir o burnout:** O principal remédio contra o burnout é o **Sono** (-0.49). Alunos que dormem mais têm níveis de esgotamento drasticamente menores. Fazer **Exercícios Físicos** (-0.15) também atua como um fator de proteção importante.
> 
> **Conclusão de Negócio:** Se uma instituição de ensino, por exemplo, quiser melhorar as notas gerais de seus alunos, a melhor estratégia não é forçar mais horas de estudo bruto, mas sim organizar melhor o calendário de provas (evitando sobreposição de prazos) e incentivar a higiene do sono e práticas esportivas.